jsonschema-use-cases.xml

```diff
@@ -0,0 +1,300 @@
+        </author>
+
+        <date year="2022"/>
+        <workgroup>JSON Schema</workgroup>
+        <keyword>JSON</keyword>
+        <keyword>Schema</keyword>
+        <keyword>Hyper Schema</keyword>
+        <keyword>Hypermedia</keyword>
+
+        <abstract>
+            <t>
+                To foster development of JSON Schema, this document contains a list of use cases and requirements that may be used to inform its development and evolution. All examples should be realistic examples of how people may use JSON Schema.
+            </t>
+        </abstract>
+        <note title="Note to Readers">
+            <t>
+                The issues list for this document can be found at
+                <eref target="https://github.com/json-schema-org/json-schema-spec/issues"/>.
+            </t>
+            <t>
+                For additional information, see <eref target="https://json-schema.org/"/>.
+            </t>
+            <t>
+                To provide feedback, use this issue tracker, the communication methods listed on the
+                homepage, or email the document editors.
+            </t>
+        </note>
+    </front>
+
+    <middle>
+        <section title="Scope and Motivation">
+            <t>
+                JSON Schema is a JSON media type for defining the structure of JSON data. JSON Schema
+                is intended to define validation, documentation, hyperlink navigation, and interaction
+                control of JSON data.
+            </t>
+            <t>
+                This document elaborates in detail what this means, and the specific use cases that shall be supported.
+            </t>
+        </section>
+
+        <section title="Conventions and Terminology">
+            <t>
+                The terms "JSON", "JSON text", "JSON value", "member", "element", "object", "array",
+                "number", "string", "boolean", "true", "false", and "null" in this document are to
+                be interpreted as defined in <xref target="RFC8259">RFC 8259</xref>.
+            </t>
+
+            <t>
+                Use Cases describe the system's behavior under various conditions, cataloging who does what with the system, for what purpose, but without concern for system design or implementation.
+            </t>
+
+            <t>
+                Requirements list functional and non-functional or quality requirements, and the use cases they may be derived from/related to.
+            </t>
+        </section>
+
+        <section title="Use Cases">
+            <section title="Structural validation" anchor="uc-structural">
+                <t>
+                    The most basic function of validating JSON is the ability to express any subset of JSON that can be described by a deterministic context-free grammar, except for distinguishing between different encodings of the same value. Schemas that limit themselves to this functionality may be compiled to a deterministic pushdown automata.
```

Comment:
> Schemas that limit themselves to this functionality may be compiled to a deterministic pushdown automata.

While (potentially) true, is it necessary to mention this?  This seems like an analysis/theory topic, and doesn't add to the definition of "validation."

My point is saying this is that while "pushdown automata" may be a valid implementation approach, it's not universal.  Mentioning it here may only serve to invalidate implementations which have a more procedural or functional architecture.

Reply:
This is a good suggestion, the way it's worded does nothing to explain why this is meaningful. I pushed an update that I think is an improvement.

Though, this isn't intended for the general public, but more as a mission statement for editors and authors to follow, and maybe for validator implementers to contribute to. So of any document anywhere, I'm least worried about jargon in this document.